features/api_packages.feature

```diff
@@ -21,16 +21,18 @@ Feature: Package related API endpoints
         # Install some outdated package
         And I run `apt install <package>=<outdated_version> -y --allow-downgrades` with sudo
         # See the update there
-        When I run `pro api u.pro.packages.updates.v1` as non-root
+        When I store candidate version of package `<package>`
+        And I regexify `candidate` stored var
+        And I run `pro api u.pro.packages.updates.v1` as non-root
         Then stdout matches regexp:
         """
-        {"download_size": \d+, "origin": ".+", "package": "<package>", "provided_by": "<provided_by>", "status": "upgrade_available", "version": "<candidate_version>"}
+        {"download_size": \d+, "origin": ".+", "package": "<package>", "provided_by": "<provided_by>", "status": "upgrade_available", "version": "$stored_vars{candidate}"}
```

Comment:
Sorry @lucasmoura the main comment of my review didn't make it past the flaky wifi I was on yesterday.

I think we can extend process_template_vars for stored_vars instead of adding a new special replacement.

This line would look like
```suggestion
        {"download_size": \d+, "origin": ".+", "package": "<package>", "provided_by": "<provided_by>", "status": "upgrade_available", "version": "$behave_var{stored_var candidate}"}
```

And then we'd add something like the following to `process_template_vars`:
```python
         elif function_name == "stored_var":
            processed_template = _replace_and_log(
                processed_template,
                match.group(0),
                context.stored_vars.get(args[1]),
                logger_fn,
            )
```

Reply:
Fair enough, I missed we call the `process_template_vars` outside the `before_scenario` setup. I will update the PR then